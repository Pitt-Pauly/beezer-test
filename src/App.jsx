import React, { Component } from 'react';
import styled, { keyframes } from 'styled-components';
import logo from './logo.svg';
import UserTable from './UserTable';

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Logo = styled.img`
  animation: ${rotate360} infinite 20s linear;
  height: 80px;
`;

const Header = styled.header`
  display: flex;
  align-items: center;
  background-color: #222;
  height: 100px;
  padding: 20px;
  color: white;
  width: 100%;
`;

const App = () => (
  <div className="App">
    <Header className="App-header">
      <Logo src={logo} className="App-logo" alt="logo" />
      <h1 className="App-title">Beezer Coding Test</h1>
    </Header>
    <UserTable />
  </div>
);

export default App;
