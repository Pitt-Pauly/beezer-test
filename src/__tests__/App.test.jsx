import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme';

test('App renders correctly', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
