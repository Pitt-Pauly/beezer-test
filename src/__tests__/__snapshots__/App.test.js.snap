// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.header
          className="App-header"
>
          <styled.img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h1
                    className="App-title"
          >
                    Beezer Coding Test
          </h1>
</styled.header>,
        <UserTable />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h1
              className="App-title"
>
              Beezer Coding Test
</h1>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Beezer Coding Test",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Beezer Coding Test",
            "type": "h1",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.header
            className="App-header"
>
            <styled.img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h1
                        className="App-title"
            >
                        Beezer Coding Test
            </h1>
</styled.header>,
          <UserTable />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h1
                className="App-title"
>
                Beezer Coding Test
</h1>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Beezer Coding Test",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "Beezer Coding Test",
              "type": "h1",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
